client/network/src/lib.rs

```diff
@@ -454,3 +454,92 @@ async fn wait_to_announce<Block: BlockT>(
 		);
 	}
 }
+
+pub async fn warp_sync_get<B, RCInterface>(
+	para_id: ParaId,
+	relay_chain_interface: RCInterface,
+	spawner: SpawnTaskHandle,
+) -> Result<oneshot::Receiver<<B as BlockT>::Header>, BoxedError>
+where
+	B: BlockT + 'static,
+	RCInterface: RelayChainInterface + 'static,
+{
+	let (sender, receiver) = oneshot::channel::<B::Header>();
+	spawner.spawn(
+		"cumulus-parachain-wait-for-target-block",
+		None,
+		async move {
+			tracing::debug!(
+				target: "cumulus-network",
+				"waiting for announce block in a background task...",
+			);
+			tracing::debug!(target: LOG_TARGET, "waiting for target block in a background task...",);
+			wait_for_target_block::<B, RCInterface>(sender, para_id, relay_chain_interface)
+				.await
+				.map_err(|e| {
+					tracing::error!(target: LOG_TARGET, "Unable to determine sync status. {}", e)
+				})
+				.unwrap();
+			tracing::debug!(target: LOG_TARGET, "target block reached",);
+		}
+		.boxed(),
+	);
+
+	return Ok(receiver)
+}
+
+async fn wait_for_target_block<B, RCInterface>(
+	sender: oneshot::Sender<<B as BlockT>::Header>,
+	para_id: ParaId,
+	relay_chain_interface: RCInterface,
+) -> Result<(), BoxedError>
+where
+	B: BlockT + 'static,
+	RCInterface: RelayChainInterface + 'static,
+{
+	match relay_chain_interface.import_notification_stream().await {
+		Ok(_) => loop {
+			let is_syncing = relay_chain_interface
+				.is_major_syncing()
+				.await
+				.map_err(|e| {
+					tracing::error!(target: LOG_TARGET, "Unable to determine sync status. {}", e)
+				})
+				.unwrap_or(false);
+
+			if !is_syncing {
+				let relay_chain_best_hash = relay_chain_interface
+					.best_block_hash()
+					.await
+					.map_err(|e| Box::new(e) as Box<_>)?;
+
+				let validation_data = relay_chain_interface
+					.persisted_validation_data(
+						relay_chain_best_hash,
+						para_id,
+						OccupiedCoreAssumption::TimedOut,
+					)
+					.await
+					.map_err(|e| Box::new(BlockAnnounceError(format!("{:?}", e))) as Box<_>)?
+					.ok_or_else(|| {
+						Box::new(BlockAnnounceError(
+							"Could not find parachain head in relay chain".into(),
+						)) as Box<_>
+					})?;
+				let target_header = B::Header::decode(&mut &validation_data.parent_head.0[..])
+					.map_err(|e| {
+						Box::new(BlockAnnounceError(format!(
+							"Failed to decode parachain head: {:?}",
+							e
+						))) as Box<_>
+					})?;
+
+				let _ = sender.send(target_header);
+				return Ok(())
+			}
+			tracing::info!(target: LOG_TARGET, "waiting for relay chain sync to complete......",);
+			sleep(std::time::Duration::from_secs(120)).await;
```

Comment:
```suggestion
```

1. The informant will already show this.
2. Sleeping 2 minutes is way too much.